pkg/promtail/scrape/scrape.go

```diff
@@ -15,10 +18,27 @@ type Config struct {
 	JobName                string                           `yaml:"job_name,omitempty"`
 	EntryParser            api.EntryParser                  `yaml:"entry_parser"`
 	PipelineStages         stages.PipelineStages            `yaml:"pipeline_stages,omitempty"`
+	JournalConfig          *JournalTargetConfig             `yaml:"journal,omitempty"`
 	RelabelConfigs         []*relabel.Config                `yaml:"relabel_configs,omitempty"`
 	ServiceDiscoveryConfig sd_config.ServiceDiscoveryConfig `yaml:",inline"`
 }
 
+// JournalTargetConfig describes systemd journal records to scrape.
+type JournalTargetConfig struct {
+	// Since holds the time after time.Now() to start retrieving records
+	// for. If 0, records will be retrieved from the earliest entry in
+	// the journal.
+	Since time.Duration `yaml:"since,omitempty"`
+
+	// Labels optionally holds labels to associate with each record coming out
+	// of the journal.
+	Labels model.LabelSet `yaml:"labels"`
+
+	// Path to a directory to read journal entries from. Defaults to system path
+	// if empty.
+	Path string `yaml:"path"`
```

Comment:
should the socket path: `/var/run/systemd/journal/socket` also be configurable too? Is it possible to be in a different place for a different OS?

Reply:
Looks like the socket is only used for writing to the journal (and the path is hard-coded). I'll update the docs to remove the bind mount to the socket.